blocks.js

```diff
@@ -3000,6 +3068,129 @@ BlockMorph.prototype.codeMappingHeader = function () {
     return hat;
 };
 
+// Variable refactoring
+
+BlockMorph.prototype.refactorThisVar = function (justTheTemplate) {
+    // Rename all occurrences of the variable this block is holding,
+    // taking care of its lexical scope
+
+    var myself = this,
+        oldName = this.blockSpec,
+        receiver = this.receiver(),
+        ide = this.parentThatIsA(IDE_Morph),
+        stage = ide.stage,
+        oldWatcher = receiver.findVariableWatcher(oldName),
+        oldValue, newWatcher;
+
+    new DialogBoxMorph(
+            this,
+            function (newName) {
+                var definer;
+                
+                if (this.parent instanceof SyntaxElementMorph) {
```

Comment:
I thought it should, since we've been playing with this, but I didn't really understand why. :)